nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -497,6 +498,27 @@ public Description matchAssignment(AssignmentTree tree, VisitorState state) {
       GenericsChecks.checkTypeParameterNullnessForAssignability(tree, this, state);
     }
 
+    if (config.isJSpecifyMode() && tree.getVariable() instanceof ArrayAccessTree) {
+      // check for a write of a @Nullable value into @NonNull array contents
+      ArrayAccessTree arrayAccess = (ArrayAccessTree) tree.getVariable();
+      ExpressionTree arrayExpr = arrayAccess.getExpression();
+      ExpressionTree expression = tree.getExpression();
+      Symbol arraySymbol = ASTHelpers.getSymbol(arrayExpr);
+      if (arraySymbol != null) {
+        boolean isElementNullable = isArrayElementNullable(arraySymbol, config);
+        if (!isElementNullable && mayBeNullExpr(state, expression)) {
+          final String message = "Writing @Nullable expression into array with @NonNull contents.";
+          ErrorMessage errorMessage = new ErrorMessage(MessageTypes.ASSIGN_ARRAY_NULLABLE, message);
+          return errorBuilder.createErrorDescription(
+              errorMessage,
+              expression,
+              buildDescription(expression),
```

Comment:
Based on the code further down, I think this should be `buildDescription(tree)`?  This controls the location at which the error is reported, so it's important.

Reply:
Fixed!